sqlitecluster/SQLiteNode.cpp

```diff
@@ -6,8 +6,15 @@
 #include <libstuff/SRandom.h>
 #include <libstuff/SQResult.h>
 #include <sqlitecluster/SQLiteCommand.h>
+#include <sqlitecluster/SQLitePeer.h>
 #include <sqlitecluster/SQLiteServer.h>
 
+// Convenience class for maintaining connections with a mesh of peers
+#define PDEBUG(_MSG_) SDEBUG("->{" << peer->name << "} " << _MSG_)
+#define PINFO(_MSG_) SINFO("->{" << peer->name << "} " << _MSG_)
+#define PHMMM(_MSG_) SHMMM("->{" << peer->name << "} " << _MSG_)
+#define PWARN(_MSG_) SWARN("->{" << peer->name << "} " << _MSG_)
```

Comment:
Where is `peer` defined? Why do these go here instead of `SQLitePeer.h`?

Reply:
I hate these. These are just C macros, so `peer` is just defined wherever it's defined. if these are used in a place where `peer` is in scope, they work. Otherwise, they don't.

For anyone unfamiliar with C macros, it's just text replacement into source code. Using the above macros and the following code:

```
void SQLiteNode::someMethod() {
    auto peer = _peerList.front();
    PINFO("lol, pinfo");
}
```

The preprocessor* just re-writes this to:
```
void SQLiteNode::someMethod() {
    auto peer = _peerList.front();
    SINFO("->{" << peer->name << "} " << "lol, pinfo");
}
````

So if your code is, instead:
```
void SQLiteNode::someMethod() {
    PINFO("lol, pinfo");
}
```

The compiler expands it to:
```
void SQLiteNode::someMethod() {
    SINFO("->{" << peer->name << "} " << "lol, pinfo");
}
```

And you get a compile-time error indicating that `peer` is undefined.

You can't easily move this into a function, either, because if you do something like this:
```
void SQLiteNode::peerInfoLog(SQLitePeer* peer, condt string& message) {
    SINFO("->{" << peer->name << "} " << message);
}
```

Because if you do, then all your log lines that used to look like:
```
2022-05-02T23:08:35.033028+00:00 expensidev2004 bedrock10001: xxxxxx (SQLiteNode.cpp:2004) _changeState [sync] [info] {cluster_node_0/SEARCHING} [NOTIFY] setting commit count to: 0
```

Now look like:
```
2022-05-02T23:08:35.033028+00:00 expensidev2004 bedrock10001: xxxxxx (SQLiteNode.cpp:1234) peerInfoLog [sync] [info] {cluster_node_0/SEARCHING} [NOTIFY] setting commit count to: 0
```

And literally every log line comes from that line in that function. Because C macros.

I'd rather just expand them out, and just write `SINFO("->{" << peer->name << "} " << _MSG_)` wherever we use these, or drop the extra prefix altogether (it's barely useful), but then I need to change every log line that uses them, so we could do that one day.

As to why it's here, it's because it's not used in `SQLiteNode.h` so by moving them here, they're at least only in scope for this file, instead of every file that includes `SQLiteNode.h`, but they're not in `SQLitePeer.h` because `SQLitePeer` doesn't (and won't) use them.

*As an aside, do you know what the unix `cpp` command is? Did you guess the C++ compiler? No, it's the C preprocessor.